### ISSUE TITLE
$JAVA_HOME env var set to $CONDA_PREFIX/Library

### ISSUE THREAD

**Issue by raybellwaves** [nils-braun/dask-sql#82]
When doing `from dask_sql import Context` it brings up `UserWarning: You are running in a conda environment, but the JAVA_PATH is not using it. If this is by mistake, set $JAVA_HOME to $CONDA_PREFIX.`. which comes from https://github.com/nils-braun/dask-sql/blob/02e2dad28741e4da84b3a957c0eedee40e262431/dask_sql/utils.py#L31

However, on the command line I see

```
>>> import os
>>> os.environ["CONDA_PREFIX"]
'C:\\Users\\131416\\AppData\\Local\\Continuum\\anaconda3\\envs\\ds-rm-obr-dev'
>>> os.environ["JAVA_HOME"]
'C:\\Users\\131416\\AppData\\Local\\Continuum\\anaconda3\\envs\\ds-rm-obr-dev\\Library'
>>> from dask_sql import Context
C:\Users\131416\AppData\Local\Continuum\anaconda3\envs\ds-rm-obr-dev\lib\site-packages\dask_sql\utils.py:30: UserWarning: You are running in a conda environment, but the JAVA_PATH is not using it. If this is by mistake, set $JAVA_HOME to $CONDA_PREFIX.
  warnings.warn(
```

Perhaps the check in https://github.com/nils-braun/dask-sql/blob/main/dask_sql/utils.py#L29 should be something like os.environ["JAVA_HOME"] != os.environ["CONDA_PREFIX"] + 'Library'

Note: i'm also on windows hence the \\

**Comment by nils-braun** [comment]
Oh interesting!
Thank you very much for reporting.

On Linux, it seems that the JAVA_HOME is set to the conda prefix - but on Windows it seems to be different.
That should be fixed, definitely

**Comment by nils-braun** [comment]
I think I have solved it with the PR (at least in the tests). If you still have a problem feel free to reopen.

**Comment by timhdesilva** [comment]
Hi! I'm having a similar problem. Is there anything I can do to fix it? Thanks!

Python 3.7.4 (default, Aug  9 2019, 18:34:13) [MSC v.1915 64 bit (AMD64)]
Type "copyright", "credits" or "license" for more information.

IPython 7.8.0 -- An enhanced Interactive Python.

>>> import dask_sql
C:\Users\tdesilva\AppData\Local\Continuum\anaconda3\lib\site-packages\dask_sql\java.py:40: UserWarning: You are running in a conda environment, but the JAVA_PATH is not using it. If this is by mistake, set $JAVA_HOME to C:\Users\tdesilva\AppData\Local\Continuum\anaconda3\Library, instead of C:\Program Files\Java\jre1.8.0_281\bin\server\.
  "You are running in a conda environment, but the JAVA_PATH is not using it. "

**Comment by nils-braun** [comment]
Hi @timhdesilva! Thanks for your issue.

In principle, this is just a warning (no exception), so the question would be: is dask-sql still working for you? 

How did you install dask-sql? Via conda or via pip? ANd do you manually set your JAVA_PATH variable or did you never touch this one?

If you are using conda, dask-sql assumes that you also installed dask-sql via conda. In this case it is very unlikely you would like to use your system java, but more likely the java installed via conda. That is why dask-sql checks for this and prints this warning. If your system java however works and is in the right version (8 <= version <= 11), you can probably ignore the warning.

**Comment by timhdesilva** [comment]
Thanks for the response! Dask-sql is working for me, so maybe I should not mess with anything haha. I installed via conda and have never touched the JAVA_PATH variable.

There's actually one other thing I'm having an issue with, and I haven't been able to find a solution online. So I have a large dataset (50GB) that needs to be merged with a small dataset that is a Pandas dataframe. Prior to the merge, I need to perform a groupby observation on the large dataset. Using Dask, I have been able to perform the groupby observation on the large dataset (which is a Dask dataframe). When I then merge the two datasets using X.merge(Y), I have no issues. The problem is that I need to perform a merge than is not exact (i.e. one column between two others), which is why I'm turning to dask-sql. When I try to do the merge with dask-sql though, I get a memory error (the number of observations should only be ~ 10x than the exact merge, so memory shouldn't be a problem).

Any ideas here? I'm thinking somehow the issue might be that I am performing a groupby operation on the Dask dataframe prior to the dask-sql merge. Is this allowed - i.e. can one do a groupby and not execute it prior to using the dask-sql create_table() command and then performing a dask-sql merge with c.sql?

Tim

________________________________
